### PR TITLE
Fix missing run and pipeline id when buttons are clicked before content load

### PR DESCRIPTION
**Pull request by drewbutlerbb4**
Fixes an issue related to [#2488](https://github.com/kubeflow/pipelines/issues/2488).
If you click certain buttons before the full page has loaded then there will be adverse results due to no pipeline/run id being specified. An example is clicking the "Delete" button on the pipeline details page before the page has fully loaded will result in receiving a popup saying "Delete succeeded for this pipeline", but nothing will be deleted.

In the pipeline details page the "Create Experiment" and "Delete" buttons were changed to include the pipeline id.
In the run details page the "Retry", "Clone Run", and "Terminate" buttons were changed to include the run id.

My only concern is that I am not sure if the api is fail safe in regards to being asked to terminate a run for a run id which is already finished running. This is possible because the Terminate button is enabled on page load and now also has the run id on page load. If this is undesirable the Terminate button can be disabled until the page loads.

This also fixes a small typo in the mock-api. Apologies for the spam.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2575)

### CODE REVIEW

**Comment by drewbutlerbb4** [comment]
cc @animeshsingh

File: frontend/mock-backend/mock-api-middleware.ts [hunk 1]
```diff
@@ -175,7 +175,7 @@ export default (app: express.Application) => {
   app.post(v1beta1Prefix + '/experiments', (req, res) => {
     const experiment: ApiExperiment = req.body;
     if (fixedData.experiments.find(e => e.name!.toLowerCase() === experiment.name!.toLowerCase())) {
-      res.status(404).send('An experiment with teh same name already exists');
+      res.status(404).send('An experiment with the same name already exists');
```
Comment (Bobgy) [inline frontend/mock-backend/mock-api-middleware.ts]: Thanks for the catch!

**Comment by Bobgy** [comment]
go module is flaky
/test kubeflow-pipeline-e2e-test

**Comment by drewbutlerbb4** [comment]
Sure, I will take a look into it. Thanks for the review!

**Pull request closed without merging**